lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_playbook_method.rb

```diff
@@ -95,15 +95,23 @@ def mark_for_retry(task_id)
 
     def manageiq_env
       {
-        'api_token'          => Api::UserTokenService.new.generate_token(@workspace.ae_user.userid, 'api'),
-        'api_url'            => MiqRegion.my_region.remote_ws_url,
+        'api_token'          => api_token,
+        'api_url'            => api_url,
         'user'               => @workspace.ae_user.href_slug,
         'group'              => @workspace.ae_user.current_group.href_slug,
         'automate_workspace' => @aw.href_slug,
         'X_MIQ_Group'        => @workspace.ae_user.current_group.description
       }
     end
 
+    def manageiq_connection_env
+      {
+        'token'       => api_token,
+        'url'         => api_url,
+        'X_MIQ_Group' => @workspace.ae_user.current_group.description
```

Comment:
Since it is being passed via extra_vars - e.g. `:extra_vars[:manageiq_connection]` the action plugin handles what gets passed down to the module, so the `X_MIQ_Group` key in both `:manageiq` and `:manageiq_connection` isn't needed. 

For consistency I would recommend just leaving it in `:extra_vars[:manageiq]`  Like you said, the ability to use 'X_MIQ_Group' to be used by the pyhon-api-client still needs to be added, pulling this option out of `:extra_vars[:manageiq_connection]` doesn't affect that change either way.

Reply:
@syncrou 
Its needed for backward compatibility for playbooks that make direct REST API calls, we can remove all of them as a set in the next release